narwhals/_arrow/series.py

```diff
@@ -359,7 +359,7 @@ def alias(self, name: str) -> Self:
         )
 
     @property
-    def dtype(self) -> DType:
+    def dtype(self: Self) -> DType:
```

Comment:
I don't believe it's needed

```suggestion
    def dtype(self) -> DType:
```

But if it is... Can you please teach me why? Thank you

Reply:
I took this as opportunity to go a bit banana into hints for the rest of the codebase (uncommitted here). When we know that `self: Self`, then `self._native_series` is known to be a `pa.ChunkedArray`, which has a `type` property.

Probably some editors do not need it anyway 😇

Reply:
sounds like something which editors should fix 😄 but in the meantime, if this helps the development workflow, then it seems like a good idea to add it 👍